src/components/register/register.html

```diff
@@ -1,69 +1,66 @@
         </div>
-
-         <div>
-                <div style="display:flex; justify-content:space-between">
-						<h5>Password *</h5>
-            <h6 *ngIf="checkError('invalid_passwd') && !checkError('passwd')" class="register-error">invalid password</h6>
-            <h6 *ngIf="checkError('no_match')" class="register-error">passwords don't match</h6>
-            <h6 *ngIf="checkError('passwd')" class="register-error">enter a password</h6>
-                </div>
-							<input  type="password" name="passwd" [(ngModel)]="user.passwd">
-				 </div>
-
-         <div>
-                <div style="display:flex; justify-content:space-between">
-						<h5>Repeat Password</h5>
-            <h6 *ngIf="checkError('passwd2')" class="register-error">enter a password</h6>
-                </div>
-							<input  type="password" name="passwd" [(ngModel)]="user.passwd2">
-				 </div>
-
-          <div id="button-container">
-				      <button ion-button color="buttonBlue" (click)="submitRegistration()">Register</button>
+        <div>
+          <div>
+            <h5>Password *</h5>
+            <p *ngIf="checkError('passwd'); else pass" class="register-error" id="registration-password-blank">enter a password</p>
+            <ng-template #pass>
+              <p *ngIf="checkError('invalid_passwd')" class="register-error" id="registration-password-invalid">enter a valid password</p>
+            </ng-template>
+            <p *ngIf="checkError('no_match')" class="register-error" id="registration-password-duplicate">passwords don't match</p>
           </div>
-
-          <div id="password-rules">
-                  <h6 *ngIf="checkError('invalid_passwd')" style="color:red">* passwords must be at least 10 characters in length, with at least one digit, special character, and uppercase letter.  </h6>
-                  <h6 *ngIf="!checkError('invalid_passwd')"> * passwords must be at least 10 characters in length, with at least one digit, special character, and uppercase letter.  </h6>
+          <input type="password" name="passwd" [(ngModel)]="user.passwd" id="pw1">
+        </div>
+        <div>
+          <div>
+            <h5>Repeat Password</h5>
+            <p *ngIf="checkError('passwd2')" class="register-error" id="registration-password2-blank">enter a password</p>
           </div>
+          <input type="password" name="passwd" [(ngModel)]="user.passwd2" id="pw2">
+        </div>
+        <div id="button-container">
+          <button ion-button color="buttonBlue" (click)="submitRegistration()">Register</button>
+        </div>
+        <div id="password-rules">
+          <p *ngIf="checkError('invalid_passwd')" style="color:red">* passwords must be at least 10 characters in
+            length, with at least one digit, special character, and uppercase letter. </p>
+          <p *ngIf="!checkError('invalid_passwd')"> * passwords must be at least 10 characters in length, with at least
```

Comment:
Think angular lets you switch styles based on condition here but it's nitpicky - don't need to spend more time on this unless you choose to

Reply:
Tried to fix, I'll have to come back to this